File: 8kyu/(8 kyu) altERnaTIng cAsE = ALTerNAtiNG CaSe/(8 kyu) altERnaTIng cAsE = ALTerNAtiNG CaSe.go
package kata

func ToAlternatingCase(str string) string {
  rs := make([]rune, len(str))
  for i, c := range str {
    r := c
    if c >= 'a' && c <= 'z' {
      r = c + 'A' - 'a'
    }
    if c >= 'A' && c <= 'Z' {
      r = c + 'a' - 'A'
    }
    rs[i] = r
  }
  return string(rs[:])
}

File: 8kyu/(8 kyu) altERnaTIng cAsE = ALTerNAtiNG CaSe/(8 kyu) altERnaTIng cAsE = ALTerNAtiNG CaSe.dart
String toAlternatingCase(final String str) {
  return String.fromCharCodes(
    str.runes.map(
      (int rune) {
        if (rune >= 'a'.codeUnitAt(0) && rune <= 'z'.codeUnitAt(0)) {
          return rune - 'a'.codeUnitAt(0) + 'A'.codeUnitAt(0);
        } else if (rune >= 'A'.codeUnitAt(0) && rune <= 'Z'.codeUnitAt(0)) {
          return rune - 'A'.codeUnitAt(0) + 'a'.codeUnitAt(0);
        } else {
          return rune;
        }
      },
    ),
  );
}